lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -124,6 +134,17 @@ def get_course_modes(self, obj):
             for mode in course_modes
         ]
 
+    def to_representation(self, instance: CourseEnrollment) -> 'OrderedDict':  # lint-amnesty, pylint: disable=unused-variable, line-too-long
```

Comment:
We didn't want to have a `course_progress` field for each user's enrollment by default. So, we have made a decision to add `course_progress` field into each user's enrollment only when `course_progress` is passed into `requested_fields` query param.  

So if we were to add this field `course_progress` as serializer field - we would have this field in every version (v0.5, v1, v2, v3, v4) of this API. We would receive something like this: "course_progress": null. We didn't want to change the response for older versions of the API, so we added this field as an additional one to  `to_representation` method. 

For `primary` user enrollment we added `course_progress` as serializer field. Because this is new functionality and we don't change old functionality.